tests/usm/usm_api.h

```diff
@@ -0,0 +1,688 @@
+/** @brief Provides test logic for the mem_advise() method tests
+ */
+template <typename T, size_t count>
+class mem_advise {
+ public:
+  static constexpr size_t size = count * sizeof(T);
+
+  mem_advise(sycl::queue &, int) {}
+
+  /** @brief The mem_advise() method doesn't support non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return false; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "mem_advise using " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    const int advice = 0;  // Reset to defaults according to the SYCL 2020 spec
+    return parent.mem_advise(ptr, size, advice,
+                             std::forward<depEventsT>(events)...);
+  }
+
+  void check(const T *) const {
+    // Results are implementation-defined
+  }
+};
+
+}  // namespace tests
+
+/** @brief Kernel name for event_generator::init() method
+ */
+template <typename T, size_t gen_buf_size>
+struct init_kernel_name;
+
+/** @brief Kernel name for event_generator::copy_arrays() method
+ */
+template <typename T, size_t gen_buf_size>
+struct copy_arrays_kernel_name;
+
+/** @brief The number of events passed to the function being checked.
+ */
+constexpr auto multiple_events = 10_events;
+
+/** @brief Helper class providing number of elements for generators buffers if
+ *         numEvents > 0
+ */
+template <size_t numEvents>
+struct gen_buf_size_selector {
+  static constexpr size_t value = 10000;
+};
+
+/** @brief Helper class specialization providing number of elements for
+ *         generators buffers if numEvents == 0
+ */
+template <>
+struct gen_buf_size_selector<0> {
+  static constexpr size_t value = 0;
+};
+
+/** @brief Helper class for generation of 'sycl::event's by calling
+ *         queue::submit with long operation in kernel execution.
+ *  @tparam buf_size Size of arrays to process
+ */
+template <typename T, size_t buf_size>
+class event_generator {
+  sycl::range<1> rng{buf_size};
+  std::array<T, buf_size> arr_src;
+  std::array<T, buf_size> arr_dst;
+  sycl::buffer<T, 1> buf_src;
+  sycl::buffer<T, 1> buf_dst;
+
+ public:
+  event_generator()
+      : rng{buf_size},
+        buf_src{arr_src.data(), rng},
+        buf_dst{arr_dst.data(), rng} {
+    arr_src.fill(T{0});
+    arr_dst.fill(T{0});
+  }
+
+  /** @brief Initialize arr_src with init_value and return sycl::event of
+   *         queue::submit()
+   *  @param value Some non-zero (non-default) value of type T
+   */
+  sycl::event init(sycl::queue &queue, T value) {
+    return queue.submit([&](sycl::handler &cgh) {
+      auto acc_src = buf_src.template get_access<sycl::access_mode::write>(cgh);
+      // single_task is used to make process long enough for testing purpose
+      // The function being tested must wait for this task to complete.
+      cgh.single_task<init_kernel_name<T, buf_size>>([=]() {
```

Comment:
```suggestion
      cgh.single_task<init_kernel_name<T, buf_size>>([=] {
```

Reply:
I've applied this comment in this commit: 6567ea1d22a06ef5b92a625523a8f336ecf0779c